.circleci/cimodel/data/simple/docker_definitions.py

```diff
@@ -0,0 +1,41 @@
+from collections import OrderedDict
+
+from cimodel.lib.miniutils import quote
+
+
+# TODO: make this generated from a matrix rather than just a static list
+IMAGE_NAMES = [
+    "pytorch-linux-bionic-cuda10.2-cudnn7-py3.8-gcc9",
+    "pytorch-linux-bionic-py3.6-clang9",
+    "pytorch-linux-bionic-cuda10.2-cudnn7-py3.6-clang9",
+    "pytorch-linux-bionic-py3.8-gcc9",
+    "pytorch-linux-xenial-cuda10-cudnn7-py3-gcc7",
+    "pytorch-linux-xenial-cuda10.1-cudnn7-py3-gcc7",
+    "pytorch-linux-xenial-cuda10.2-cudnn7-py3-gcc7",
+    "pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4",
+    "pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc7",
+    "pytorch-linux-xenial-py3-clang5-android-ndk-r19c",
+    "pytorch-linux-xenial-py3-clang5-asan",
+    "pytorch-linux-xenial-py3.8",
+    "pytorch-linux-xenial-py3.6-clang7",
+    "pytorch-linux-xenial-py3.6-gcc4.8",
+    "pytorch-linux-xenial-py3.6-gcc5.4",
+    "pytorch-linux-xenial-py3.6-gcc7.2",
+    "pytorch-linux-xenial-py3.6-gcc7",
+    "pytorch-linux-xenial-pynightly",
+    "pytorch-linux-xenial-rocm3.3-py3.6",
+]
```

Comment:
How about something like that, to avoid string duplication:
```suggestion
def _bionic(suffix):
  return "pytorch-linux-bionic-" + suffix
  
def _xenial(suffix):
  return "pytorch-linux-xenial-" + suffix

    
IMAGE_NAMES = [
    _bionic("cuda10.2-cudnn7-py3.8-gcc9"),
    _bionic("py3.6-clang9"),
    _bionic("cuda10.2-cudnn7-py3.6-clang9"),
    _bionic("py3.8-gcc9"),
    _xenial("cuda10.1-cudnn7-py3-gcc7"),
    _xenial("cuda10.2-cudnn7-py3-gcc7"),
    _xenial("cuda9.2-cudnn7-py3-gcc5.4"),
    _xenial("cuda9.2-cudnn7-py3-gcc7"),
    _xenial("py3-clang5-android-ndk-r19c"),
    _xenial("py3-clang5-asan"),
    _xenial("py3.8"),
    _xenial("py3.6-clang7"),
    _xenial("py3.6-gcc4.8"),
    _xenial("py3.6-gcc5.4"),
    _xenial("py3.6-gcc7.2"),
    _xenial("py3.6-gcc7"),
    _xenial("pynightly"),
    _xenial("rocm3.3-py3.6"),
]
```

Reply:
I wanted to keep this as simple as possible for this PR, we can decide on how to split this into better codegen later on.

Reply:
I personally like the literal strings, it makes it easier to grep